[incubator/patroni] Add kubernetes DCS

ese opened the pull request:
**What this PR does / why we need it**:

* Add kubernetes DCS, much simple way when run in kubernetes than use zookeeper or etcd
* Use kubernetes DCS as default and avoid to deploy other service
* Remove chart label in volumeClaimTemplates metadata since it is inmutable and this value change on every chart version so is preventing updates
* Add UpdateStrategy to StatefulSet

ese committed:
Add kubernetes DCS

Signed-off-by: Sergio Ballesteros <snaker@locolandia.net>

ese committed:
Set UpdateStrategy for statefulset

Signed-off-by: Sergio Ballesteros <snaker@locolandia.net>

ese committed:
Remove chart label from volumetemplate (inmutable field)

Signed-off-by: Sergio Ballesteros <snaker@locolandia.net>

ese committed:
Bump chart to 0.10.0

Signed-off-by: Sergio Ballesteros <snaker@locolandia.net>

NicolasT commented on incubator/patroni/Chart.yaml:
```diff
@@ -1,6 +1,6 @@
 name: patroni
 description: 'Highly available elephant herd: HA PostgreSQL cluster.'
-version: 0.9.0
+version: 0.9.1
```
I believe the warrants more than just a micro/patch bump.

ese replied:
Ok, changed to 0.10.0

ese committed:
Bump spilo to 1.4-p11

Signed-off-by: Sergio Ballesteros <snaker@locolandia.net>

ese committed:
Update RBAC objects to v1

Signed-off-by: Sergio Ballesteros <snaker@locolandia.net>

ese committed:
Update dependencies

Signed-off-by: Sergio Ballesteros <snaker@locolandia.net>

mnencia committed:
Fix some environment variables

 * `DCS_ENABLE_KUBERNETES_API` is required to be set to true

 * `PATRONI_KUBERNETES_LABELS` does nothing, me must use
   `KUBERNETES_LABELS` instead. It must also be a valid JSON string.

 * `KUBERNETES_SCOPE_LABEL` must be set to "release" to match what
    created by helm

Signed-off-by: Marco Nenciarini <marco.nenciarini@2ndquadrant.it>
Signed-off-by: Sergio Ballesteros <snaker@locolandia.net>
